addon/mixins/flexberry-maplayer-actions-handler.js

```diff
@@ -364,5 +370,36 @@ export default Ember.Mixin.create({
 
     Ember.set(layer, 'isDeleted', true);
     return layer;
-  }
+  },
+
+  /**
+   Sets indexes for layers hierarchy.
+
+    @method setIndexes
+    @param {Object} rootLayer Root layer of the tree.
+    @param {Boolean} isAdded True when layer is added.
+    @param {Int} index First index.
+    @returns {Int} Last index.
+  */
+  setIndexes(rootLayer, isAdded, index) {
+    let layers = rootLayer.get('layers');
+    if (Ember.isArray(layers)) {
+      if (Ember.isNone(index)) {
+        index = layers.objectAt(0).get('index') || 0;
+        index = isAdded ? index + 1 : index - 1;
+      }
+
+      layers.forEach((layer) => {
+        if (!layer.get('isDeleted')) {
+          layer.set('index', index);
+          index--;
+          if (layer.get('type') === 'group') {
```

Comment:
Instead of check to type name (which can be potentially renamed) it is better to check that layer.layers is an array.